internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -395,40 +382,57 @@ func calculateStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet
 	}
 
 	// Calculate the label selector. We check the error in the MD reconcile function, ignore here.
-	selector, _ := metav1.LabelSelectorAsSelector(&deployment.Spec.Selector)
-
-	status := clusterv1.MachineDeploymentStatus{
-		// TODO: Ensure that if we start retrying status updates, we won't pick up a new Generation value.
-		ObservedGeneration:  deployment.Generation,
-		Selector:            selector.String(),
-		Replicas:            mdutil.GetActualReplicaCountForMachineSets(allMSs),
-		UpdatedReplicas:     mdutil.GetActualReplicaCountForMachineSets([]*clusterv1.MachineSet{newMS}),
-		ReadyReplicas:       mdutil.GetReadyReplicaCountForMachineSets(allMSs),
-		AvailableReplicas:   availableReplicas,
-		UnavailableReplicas: unavailableReplicas,
-		Conditions:          deployment.Status.Conditions,
-	}
+	selector, _ := metav1.LabelSelectorAsSelector(&machineDeployment.Spec.Selector)
 
-	if *deployment.Spec.Replicas == status.ReadyReplicas {
-		status.Phase = string(clusterv1.MachineDeploymentPhaseRunning)
+	if desiredReplicas == readyReplicas {
+		phase = string(clusterv1.MachineDeploymentPhaseRunning)
+		conditions.MarkTrue(machineDeployment, clusterv1.ResizedCondition)
 	}
-	if *deployment.Spec.Replicas > status.ReadyReplicas {
-		status.Phase = string(clusterv1.MachineDeploymentPhaseScalingUp)
+	if desiredReplicas > readyReplicas {
+		phase = string(clusterv1.MachineDeploymentPhaseScalingUp)
+		conditions.MarkFalse(machineDeployment, clusterv1.ResizedCondition, clusterv1.ScalingUpReason, clusterv1.ConditionSeverityWarning,
+			"Scaling up MachineDeployment to %d replicas (actual %d)", desiredReplicas, readyReplicas)
 	}
 	// This is the same as unavailableReplicas, but we have to recalculate because unavailableReplicas
 	// would have been reset to zero above if it was negative
 	if totalReplicas-availableReplicas < 0 {
-		status.Phase = string(clusterv1.MachineDeploymentPhaseScalingDown)
+		phase = string(clusterv1.MachineDeploymentPhaseScalingDown)
+		conditions.MarkFalse(machineDeployment, clusterv1.ResizedCondition, clusterv1.ScalingDownReason, clusterv1.ConditionSeverityWarning,
+			"Scaling down MachineDeployment to %d replicas (actual %d)", desiredReplicas, availableReplicas)
```

Comment:
I wonder why actual here are availableReplicas while above they are readyReplicas. Is this correct?

Reply:
I'm honestly not sure atm, I didn't wanted to change it for now.

Reply:
Sounds reasonable :)